ros_buildfarm/status_page.py

```diff
@@ -530,6 +531,303 @@ def _get_comparable_loose_versions(version_str1, version_str2):
     return loose_version1, loose_version2
 
 
+def build_blocked_releases_page(
+        config_url, rosdistro_name,
+        output_dir, repo_names=None, copy_resources=False):
+
+    start_time = time.localtime()
+
+    repos_info = _get_blocked_releases_info(config_url, rosdistro_name, repo_names)
+    repos_data = [_format_repo_table_row(name, data) for name, data in repos_info.items()]
+
+    template_name = 'status/blocked_releases_page.html.em'
+    data = {
+        'title': 'ROS %s - blocked releases' % rosdistro_name,
+        'start_time': time.strftime('%Y-%m-%d %H:%M:%S %z', start_time),
+
+        'resource_hashes': get_resource_hashes(),
+
+        'rosdistro_name': rosdistro_name.capitalize(),
+
+        'repos_data': repos_data,
+    }
+    html = expand_template(template_name, data)
+    output_filename = os.path.join(
+        output_dir, 'blocked_releases_%s.html' % rosdistro_name)
+    print("Generating blocked releases page '%s':" % output_filename)
+    with open(output_filename, 'w') as h:
+        h.write(html)
+
+    additional_resources(output_dir, copy_resources=copy_resources)
+
+
+def _div_wrap(data, name=None):
+    if name:
+        return '<div id="{0}">{1}</div>'.format(name, data)
+    else:
+        return '<div>{0}</div>'.format(data)
```

Comment:
The main logic in both branches is the same. This could be changed into something without duplication like:

    if_attr = ' id="%s"' % name if name is not None else ''
    return '<div{0}>{1}</div>'.format(if_attr, data)